### ISSUE TITLE
Database migration can end up deleting db 

### ISSUE THREAD

**Issue by Mikescher** [louislam/uptime-kuma#2778]
### ⚠️ Please verify that this bug has NOT been raised before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/uptime-kuma/security/policy)

### Description

I upgraded my uptime-kuma docker container to 1.20.    
During the upgrade uptime-kuma failed to create a DB backup (I'm running on a small VPS with only 10GB space) du to insufficient HDD space.  
*(I think)* when trying to restore the (non-existant) backup uptime-kuma first deleted the kuma.db file, leaving me at the end with neither a backup nor the actual DB file.

In the end I restored a backup from january and only lost ~ 3 weeks of data. But it was a bit scary :D

### 👟 Reproduction steps

update uptime-kuma on a system with not enough space for a full copy of kuma.db. If there are no prior backups you end up with no DB file and no backup file.

### 👀 Expected behavior

do not delete the kuma.db file in the backup-restore process. (perhaps only rename it first and then delete it at the end of the process, after the copy suceeded...)

### 😓 Actual Behavior

kuma.db deleted an no backup.

### 🐻 Uptime-Kuma Version

1.19 -> 1.20

### 💻 Operating System and Arch

Debian GNU/Linux 11 (bullseye)

### 🐋 Docker Version

20.10.23

### 📝 Relevant log output

```shell
==> Performing startup jobs and maintenance tasks
changed ownership of '/app/data/sqlite3' from node:node to 0:0
==> Starting application with user 0 group 0
Welcome to Uptime Kuma
Your Node.js version: 16
2023-02-14T00:00:57Z [SERVER] INFO: Welcome to Uptime Kuma
2023-02-14T00:00:58Z [SERVER] INFO: Node Env: production
2023-02-14T00:00:58Z [SERVER] INFO: Importing Node libraries
2023-02-14T00:00:58Z [SERVER] INFO: Importing 3rd-party libraries
2023-02-14T00:01:03Z [SERVER] INFO: Creating express and socket.io instance
2023-02-14T00:01:03Z [SERVER] INFO: Server Type: HTTP
2023-02-14T00:01:03Z [SERVER] INFO: Importing this project modules
2023-02-14T00:01:04Z [NOTIFICATION] INFO: Prepare Notification Providers
2023-02-14T00:01:04Z [SERVER] INFO: Version: 1.20.0
2023-02-14T00:01:04Z [DB] INFO: Data Dir: ./data/
2023-02-14T00:01:04Z [SERVER] INFO: Connecting to the Database
2023-02-14T00:01:05Z [DB] INFO: SQLite config:
[ { journal_mode: 'wal' } ]
[ { cache_size: -12000 } ]
2023-02-14T00:01:05Z [DB] INFO: SQLite Version: 3.39.4
2023-02-14T00:01:05Z [SERVER] INFO: Connected
2023-02-14T00:01:05Z [DB] INFO: Your database version: 10
2023-02-14T00:01:05Z [DB] INFO: Latest database version: 10
2023-02-14T00:01:05Z [DB] INFO: Database patch not needed
2023-02-14T00:01:05Z [DB] INFO: Database Patch 2.0 Process
2023-02-14T00:01:05Z [DB] INFO: patch-ping-packet-size.sql is not patched
2023-02-14T00:01:05Z [DB] INFO: Backing up the database
2023-02-14T00:01:23Z [DB] INFO: Closing the database
2023-02-14T00:01:25Z [DB] INFO: SQLite closed
Error: ENOSPC: no space left on device, copyfile './data/kuma.db' -> './data/kuma.db.bak20230214000105'
    at Object.copyFileSync (node:fs:2847:3)
    at Function.backup (/app/server/database.js:458:16)
    at Function.patch2Recursion (/app/server/database.js:366:18)
    at Function.patch2 (/app/server/database.js:249:28)
    at async Function.patch (/app/server/database.js:226:9)
    at async initDatabase (/app/server/server.js:1656:5)
    at async /app/server/server.js:171:5 {
  errno: -28,
  syscall: 'copyfile',
  code: 'ENOSPC',
  path: './data/kuma.db',
  dest: './data/kuma.db.bak20230214000105'
}
2023-02-14T00:01:25Z [DB] ERROR: Start Uptime-Kuma failed due to issue patching the database
2023-02-14T00:01:25Z [DB] ERROR: Please submit the bug report if you still encounter the problem after restart: https://github.com/louislam/uptime-kuma/issues
2023-02-14T00:01:25Z [DB] ERROR: Patching the database failed!!! Restoring the backup
Trace: Error: ENOENT: no such file or directory, copyfile './data/kuma.db.bak20230214000105' -> './data/kuma.db'
    at Object.copyFileSync (node:fs:2847:3)
    at Function.restore (/app/server/database.js:518:16)
    at Function.patch2 (/app/server/database.js:263:18)
    at async Function.patch (/app/server/database.js:226:9)
    at async initDatabase (/app/server/server.js:1656:5)
    at async /app/server/server.js:171:5 {
  errno: -2,
  syscall: 'copyfile',
  code: 'ENOENT',
  path: './data/kuma.db.bak20230214000105',
  dest: './data/kuma.db'
}
    at process.<anonymous> (/app/server/server.js:1794:13)
    at process.emit (node:events:513:28)
    at emit (node:internal/process/promises:140:20)
    at processPromiseRejections (node:internal/process/promises:274:27)
    at processTicksAndRejections (node:internal/process/task_queues:97:32)
If you keep encountering errors, please report to https://github.com/louislam/uptime-kuma/issues
```

**Comment by Computroniks** [comment]
Yes, this seems to be the case. I will look into fixing it now.

**Comment by Computroniks** [comment]
This should have been fixed in #2779. I have added a check to make sure that all the database backup files exist before deleting the currently working database.

**Comment by bastouf** [comment]
Same here... and i lost all my setup 😭
